### PR TITLE
Cheapest clinic functionality

### PR DESCRIPTION
**Pull request by saragilford**

### CODE REVIEW

File: lib/clinic_finder.rb [hunk 1]
```diff
@@ -9,5 +9,25 @@ class ClinicFinder
   def initialize(yml_file)
 		@clinics = ::YAML.load_file(yml_file)
   end
+
+  def locate_cheapest_clinic(gestational_age:, naf_clinics_only: false)
```
Comment (colinxfleming) [inline lib/clinic_finder.rb:13]: intended semicolon after gestational age here?

File: lib/clinic_finder.rb [hunk 2]
```diff
@@ -9,5 +9,25 @@ class ClinicFinder
...
+    @helper = ::ClinicFinder::GestationHelper.new(gestational_age)
+    @gestational_tier = @helper.gestational_tier
+    decorate_data(available_clinics)
+  end
+
+  private def decorate_data(data)
+    sorted_clinics = []
+    three_cheapest(data).map { |clinic_array| sorted_clinics << { clinic_array.first => clinic_array.last } }
```
Comment (colinxfleming) [inline lib/clinic_finder.rb]: can you add a comment about what's going on here? I think I get it but having explicit would rule

File: lib/clinic_finder.rb [hunk 3]
```diff
@@ -9,5 +9,25 @@ class ClinicFinder
...
+    sorted_clinics
+  end
+
+  private def three_cheapest(data)
+    data.sort_by { |name, information| information[@gestational_tier] }.first(3)
+  end
+
+  private def available_clinics
+    @clinics.keep_if { |name, information| information[@gestational_tier] && @helper.within_gestational_limit?(information['gestational_limit']) }
```
Comment (colinxfleming) [inline lib/clinic_finder.rb:31]: `keep_if` <3 <3 <3

File: lib/clinic_finder/gestation_helper.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
+class ClinicFinder
+  class GestationHelper
+    attr_reader :gestational_age, :gestational_weeks
+
+    def initialize(gestational_age)
+      @gestational_age = gestational_age
+      @gestational_weeks = (@gestational_age/7.0).ceil
```
Comment (colinxfleming) [inline lib/clinic_finder/gestation_helper.rb]: Again, so we have it explicit, can we make a note about some of the business logic here?

File: test/test_clinic_finder.rb [hunk 5]
```diff
@@ -10,4 +10,24 @@ def setup
   def test_that_initialize_sets_clinic_variable
   	assert_kind_of Hash, @abortron.clinics
   end
+
```
Comment (colinxfleming) [inline test/test_clinic_finder.rb:13]: this looks sooooo good. If there's anything else that needs a test can you leave a comment, just so we have a record of where to add tests after the fact?

**Review by colinxfleming — APPROVED** [review]
Had a few comments and points of clarification, but 

image

File: lib/clinic_finder.rb (hunk 1 above)
Comment (saragilford) [reply lib/clinic_finder.rb:13]: @colinxfleming intentional! args are named. call as such!
```
@abortron = Abortron::ClinicFinder.new(yml_file)
@abortron.locate_cheapest_clinic(gestational_age: 50, naf_clinics_only: true)
```

**Pull request merged**